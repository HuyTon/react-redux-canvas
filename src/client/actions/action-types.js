// Canvas
export const GET_CANVAS_SUCCESS = 'GET_CANVAS_SUCCESS';

// Canvas Layout
export const LOAD_CANVAS_LAYOUT = 'LOAD_CANVAS_LAYOUT';

// Shapes
export const GET_SHAPES_SUCCESS = 'GET_SHAPES_SUCCESS';
export const DRAW_FULL_SHAPE = 'DRAW_FULL_SHAPE';
export const DRAW_SEMI_SHAPE = 'DRAW_SEMI_SHAPE';
export const REMOVE_SHAPE = 'REMOVE_SHAPE';
