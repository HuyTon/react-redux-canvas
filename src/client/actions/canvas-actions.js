import * as types from './action-types';

export const getCanvasSuccess = (canvas) => {
  return {
    type: types.GET_CANVAS_SUCCESS,
    canvas
  };
}