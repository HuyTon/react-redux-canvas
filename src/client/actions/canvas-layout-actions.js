import * as types from './action-types';

export const loadCanvasLayout = (title) => {
    return {
        type: types.LOAD_CANVAS_LAYOUT,
        title
    };
}
