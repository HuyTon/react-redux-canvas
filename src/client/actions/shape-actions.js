import * as types from './action-types';

export const getShapesSuccess = (shapes) => {
  return {
    type: types.GET_SHAPES_SUCCESS,
    shapes
  };
}

export const drawFullShape = (shapeId) => {
  return {
    type: types.DRAW_FULL_SHAPE,
    shapeId
  };
}

export const drawSemiShape = (shapeId) => {
  return {
    type: types.DRAW_SEMI_SHAPE,
    shapeId
  };
}

export const removeShape = (shapeId) => {
  return {
    type: types.REMOVE_SHAPE,
    shapeId
  };
}