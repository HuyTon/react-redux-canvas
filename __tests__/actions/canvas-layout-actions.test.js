import * as canvasLayoutActions from '../../src/client/actions/canvas-layout-actions';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore({});

describe('canvas_actions', () => {
    beforeEach(() => {
        store.clearActions();
    });
    describe('loadCanvasLayout', () => {
        test('Dispatches the correct action loadCanvasLayout', () => {
            store.dispatch(canvasLayoutActions.loadCanvasLayout('Demo Magic Canvas'));
            expect(store.getActions()).toMatchSnapshot();
        });
    });
})