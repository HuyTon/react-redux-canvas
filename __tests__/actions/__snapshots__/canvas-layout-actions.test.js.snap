// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`canvas_actions loadCanvasLayout Dispatches the correct action loadCanvasLayout 1`] = `
Array [
  Object {
    "title": "Demo Magic Canvas",
    "type": "LOAD_CANVAS_LAYOUT",
  },
]
`;
